js/src/edit-block/components/side.js

```diff
@@ -2,34 +2,58 @@
  * External dependencies
  */
 import * as React from 'react';
+import classNames from 'classnames';
 
 /**
  * WordPress dependencies
  */
+import { useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
-import { FieldPanel } from './';
+import { BlockPanel, FieldPanel } from './';
 
 /**
  * The side component.
  *
- * @return {React.ReactElement} The side component.
+ * @return {React.ReactElement} The side area.
  */
 const Side = () => {
+	const [ displayBlockPanel, setDisplayBlockPanel ] = useState( true );
+	const toggleDisplayFieldPanel = () => {
+		setDisplayBlockPanel( ( shouldDisplayFieldPanel ) => ! shouldDisplayFieldPanel );
+	};
+	const buttonClass = 'flex items-center h-12 px-5 text-sm focus:outline-none';
+
 	return (
 		<div className="side flex-shrink-0 flex flex-col border-l border-gray-300 overflow-scroll">
 			<div className="flex w-full border-b border-gray-300">
-				<button className="flex items-center h-12 px-5 text-sm focus:outline-none">
+				<button
+					onClick={ toggleDisplayFieldPanel }
+					className={
+						classNames(
+							buttonClass,
+							{ 'font-semibold border-b-4 border-blue-600': displayBlockPanel }
+						)
+					}
+				>
 					{ __( 'Block', 'genesis-custom-blocks' ) }
 				</button>
-				<button className="flex items-center h-12 px-5 text-sm font-semibold border-b-4 border-blue-600 focus:outline-none">
+				<button
+					onClick={ toggleDisplayFieldPanel }
+					className={
+						classNames(
+							buttonClass,
+							{ 'font-semibold border-b-4 border-blue-600': ! displayBlockPanel }
+						)
+					}
+				>
 					{ __( 'Field', 'genesis-custom-blocks' ) }
 				</button>
 			</div>
-			<FieldPanel />
+			{ displayBlockPanel ? <BlockPanel /> : <FieldPanel /> }
```

Comment:
Here's the `<Side>` component:

![the-panels-expl](https://user-images.githubusercontent.com/4063887/99036203-d8ca3100-2546-11eb-8e7a-c4fe4fe714fb.gif)